### DISCUSSION TITLE
Adding new depenencies to the workflows

### DISCUSSION THREAD

**Discussion by wwmm** [wwmm/easyeffects#1928] [General]
Hi @vchernin. Now that the `gsl` library package is needed the workflow configuration has to be updated. How to add it?

**Comment by wwmm** [comment]
Oh... The Arch Linux workflow uses the PKGBUILD in our repository?  I did not notice this before. By the way this pkgbuild build fails in my computer. Strange...

**Reply by wwmm** [reply]
```
==> Making package: easyeffects-git pkgvernotupdated-1 (ter 01 nov 2022 13:15:51)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
==> Extracting sources...
==> Starting pkgver()...
fatal: not a git repository (or any parent up to mount point /home/wallace)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
==> ERROR: pkgver is not allowed to be empty.
==> ERROR: pkgver() generated an invalid version: 
```
As I usually use another PKGBUILD file based on the one from AUR I did not notice this before.

**Reply by vchernin** [reply]
Yeah the PKGBUILD is the right thing to edit.  It is strange it does not build locally. Does it work if you revert the PKGBUILD changes in https://github.com/wwmm/easyeffects/commit/3b6b83e868d3475fbcba8ccdb3aee0252fe8ac8e? Some of the pkgver() logic there is in hindsight a bit excessive, but it should work.

**Reply by vchernin** [reply]
Hm is EasyEffects stored in `/home/wallace/easyeffects`?. Git was used in the PKGBUILD before and is obviously present on your system, so perhaps it is the strange `cd` logic which was strangely needed to make CI happy.

**Reply by vchernin** [reply]
I tried `makepkg -si` within the easyeffects directory in a arch vm and it worked:
```
==> Starting pkgver()...
==> Updated version: easyeffects-git 6.3.0.r280.ge0f26f9b-1
```
 maybe that is the difference? I don't know of a way to find the root directory in the PKGBUILD so git operations will always happen in the right directory.

**Reply by vchernin** [reply]
Of course it later fails to build locally when installing the files...
```python
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/mesonbuild/mesonmain.py", line 153, in run
    return options.run_func(options)
  File "/usr/lib/python3.10/site-packages/mesonbuild/minstall.py", line 758, in run
    with open(os.path.join(log_dir, 'install-log.txt'), 'w', encoding='utf-8') as lf:
PermissionError: [Errno 13] Permission denied: 'meson-private/../meson-logs/install-log.txt'
 ```

**Reply by wwmm** [reply]
>Hm is EasyEffects stored in /home/wallace/easyeffects?

No. When using the PKGBUILD I usually go to an empty folder and run `makepkg -si`. So makepkg has to download the sources first. So not having the line `source=("easyeffects::git+https://github.com/wwmm/easyeffects.git")` is the first problem. But restoring it is not enough.

**Reply by wwmm** [reply]
Maybe we should have 2 pkgbuild files. One for the workflow and another for general usage.

**Reply by vchernin** [reply]
>  So not having the line source=("easyeffects::git+https://github.com/wwmm/easyeffects.git") is the first problem.

I see, for CI both the flatpak and the arch build try to get the local files, as we can't assume the files in master are the ones we want.

> Maybe we should have 2 pkgbuild files. One for the workflow and another for general usage.

Yeah I don't see a better way to support both use cases.

**Reply by wwmm** [reply]
I created a new pkgbuild only for local usage.
